src/test/java/guitests/guihandles/TaskCardHandle.java

```diff
@@ -0,0 +1,46 @@
+package guitests.guihandles;
+
+import guitests.GuiRobot;
+import javafx.scene.Node;
+import javafx.stage.Stage;
+import seedu.todolist.model.task.ReadOnlyTask;
+
+/**
+ * Provides a handle to a task card in the task list panel.
+ */
+public class TaskCardHandle extends GuiHandle {
+    private static final String NAME_FIELD_ID = "#name";
+
+    private Node node;
+
+    public TaskCardHandle(GuiRobot guiRobot, Stage primaryStage, Node node){
+        super(guiRobot, primaryStage, null);
+        this.node = node;
+    }
+
+    protected String getTextFromLabel(String fieldId) {
```

Comment:
Any particular reason for using `protected` ? For this application you can get away with using only public and private.